### PR TITLE
Update project dependencies

### PR DESCRIPTION
**Pull request by fabiocav**
Applying required dependency updates throughout the solution.

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
  * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/DotNetWorker.Core/DotNetWorker.Core.csproj [hunk 1]
```diff
@@ -32,5 +32,6 @@
     <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="5.0.0" />
     <PackageReference Include="System.Threading.Tasks.Extensions" Version="4.5.4" />
     <PackageReference Include="Microsoft.Extensions.Options" Version="5.0.0" />
+    <PackageReference Include="System.Text.Encodings.Web" Version="8.0.0" />
```
Comment (kshyju) [inline src/DotNetWorker.Core/DotNetWorker.Core.csproj:35]: Did you mean to change this to 8X (than 6x)?

File: src/DotNetWorker.Grpc/DotNetWorker.Grpc.csproj [hunk 2]
```diff
@@ -21,9 +21,9 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Azure.Core" Version="1.10.0" />
+    <PackageReference Include="Azure.Core" Version="1.37.0" />
     <PackageReference Include="Google.Protobuf" Version="3.23.3" />
```
Comment (kshyju) [inline src/DotNetWorker.Grpc/DotNetWorker.Grpc.csproj:25]: Should we also consider upgrading Google.Protobuf to[ 3.25.2](https://www.nuget.org/packages/Google.Protobuf/3.25.2)?

File: test/Sdk.Analyzers.Tests/Sdk.Analyzers.Tests.csproj [hunk 3]
```diff
@@ -7,19 +7,26 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.6.3" />
-    <PackageReference Include="SmartAnalyzers.RoslynTestKit" Version="4.7.0" />
-    <PackageReference Include="xunit" Version="2.4.2" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.5">
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0" />
+    <PackageReference Include="SmartAnalyzers.RoslynTestKit" Version="5.0.2" />
+    <PackageReference Include="System.Data.SqlClient" Version="4.8.6" />
```
Comment (kshyju) [inline test/Sdk.Analyzers.Tests/Sdk.Analyzers.Tests.csproj:12]: Did you mean to add `System.Data.SqlClient`? Where do we use it?

Comment (kshyju) [reply test/Sdk.Analyzers.Tests/Sdk.Analyzers.Tests.csproj:12]: Same question for `System.Drawing.Common`

File: samples/EntityFramework/EntityFramework.csproj [hunk 4]
```diff
@@ -6,16 +6,27 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http" Version="3.1.0" />
-    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.16.2" />
-    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.20.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="8.0.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="8.0.0">
+    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.16.4" />
+    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.20.1" />
+
```
Comment (kshyju) [inline samples/EntityFramework/EntityFramework.csproj:11]: extra line break.

File: samples/EntityFramework/EntityFramework.csproj [hunk 5]
```diff
@@ -6,16 +6,27 @@
...
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
-    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="8.0.0" />
-    <PackageReference Include="Newtonsoft.Json" Version="13.0.2" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="8.0.1" />
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
     <PackageReference Include="System.Net.NameResolution" Version="4.3.0" />
+
+
```
Comment (kshyju) [inline samples/EntityFramework/EntityFramework.csproj:21]: extra line break.

File: src/DotNetWorker.Core/DotNetWorker.Core.csproj (hunk 1 above)
Comment (fabiocav) [reply src/DotNetWorker.Core/DotNetWorker.Core.csproj:35]: Yes. 6.x will reach EOL very soon. Moving to 8.0.0 will avoid another major bump.

File: test/Sdk.Analyzers.Tests/Sdk.Analyzers.Tests.csproj (hunk 3 above)
Comment (fabiocav) [reply test/Sdk.Analyzers.Tests/Sdk.Analyzers.Tests.csproj:12]: Yes. Those have been promoted from transitive dependencies.

File: src/DotNetWorker.Grpc/DotNetWorker.Grpc.csproj (hunk 2 above)
Comment (fabiocav) [reply src/DotNetWorker.Grpc/DotNetWorker.Grpc.csproj:25]: Not required as part of this pass, but I can. Will leave this unresolved and if pushing an update, will bump this up as well.

**Review by satvu — APPROVED** [review]
I think the test failures are package version checks and/or values generated depending on references so tentative approval

**Review by kshyju — APPROVED** [review]

File: extensions/Worker.Extensions.Tables/release_notes.md [hunk 6]
```diff
@@ -4,6 +4,8 @@
 - My change description (#PR/#issue)
 -->
 
-### Microsoft.Azure.Functions.Worker.Extensions.Tables <version>
+### Microsoft.Azure.Functions.Worker.Extensions.Tables 1.2.2
```
Comment (mathewc) [inline extensions/Worker.Extensions.Tables/release_notes.md:7]: Should this value match the 1.3.0 version you specified below?

Comment (fabiocav) [reply extensions/Worker.Extensions.Tables/release_notes.md:7]: Yes! Will update.

File: extensions/Worker.Extensions.ServiceBus/src/Worker.Extensions.ServiceBus.csproj [hunk 7]
```diff
@@ -15,11 +15,11 @@
   <Import Project="..\..\..\build\Extensions.props" />
 
   <ItemGroup>
-    <PackageReference Include="Azure.Identity" Version="1.10.2" />
-    <PackageReference Include="Azure.Messaging.ServiceBus" Version="7.17.0" />
-    <PackageReference Include="Microsoft.Extensions.Azure" Version="1.6.3" />
-    <PackageReference Include="Google.Protobuf" Version="3.23.4" />
-    <PackageReference Include="Grpc.Tools" Version="2.56.2" PrivateAssets="All"/>
+    <PackageReference Include="Azure.Identity" Version="1.10.4" />
```
Comment (jviau) [inline extensions/Worker.Extensions.ServiceBus/src/Worker.Extensions.ServiceBus.csproj:18]: We are okay with the major version revs in the transitive dependencies here?

File: sdk/Sdk.Generators/Sdk.Generators.csproj [hunk 8]
```diff
@@ -9,16 +9,16 @@
     <RootNamespace>Microsoft.Azure.Functions.Worker.Sdk.Generators</RootNamespace>
     <IncludeBuildOutput>false</IncludeBuildOutput>
     <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
-    <MinorProductVersion>1</MinorProductVersion>
-    <PatchProductVersion>7</PatchProductVersion>
+    <MinorProductVersion>2</MinorProductVersion>
```
Comment (jviau) [inline sdk/Sdk.Generators/Sdk.Generators.csproj:12]: This technically does not need a minor version rev (only patch) as it is a private asset being updated. But minor is fine as well.

File: sdk/release_notes.md [hunk 9]
```diff
@@ -6,12 +6,9 @@
 
 ### Microsoft.Azure.Functions.Worker.Sdk <version> (meta package)
```
Comment (jviau) [inline sdk/release_notes.md:7]:
```suggestion
### Microsoft.Azure.Functions.Worker.Sdk 1.17.0-preview3 (meta package)
```

**Review by jviau — APPROVED** [review]

**Review by liliankasem — COMMENTED** [review]
storage ext looking good

**Pull request merged**